assets/js/JsonMenuEditor.js

```diff
@@ -132,128 +133,104 @@ export default class JsonMenuEditor {
+            $modal.find('.modal-loading').hide();
+            $modal.find('.ajax-content').show();
+            $modal.find('form :input').prop('disabled', false);
+            $modal.find('.btn-json-menu-nested-save').prop('disabled', false);
+            $modal.find('.btn-json-menu-nested-save').show();
+        }
+        function modalStateloading($modal) {
+            $modal.find('.modal-loading').show();
+            $modal.find('.ajax-content').html('').hide();
+            $modal.find('.btn-json-menu-nested-save').hide();
+        }
+        function modalStateSaving($modal) {
+            $modal.find('.modal-loading').show();
+            $modal.find('form :input').prop('disabled', true);
+            $modal.find('.btn-json-menu-nested-save').prop('disabled', true);
+        }
 
-            if (action === 'add') {
-                self.addItem(target, 'prototype-nested', {
-                    'icon': node.icon,
-                    'label': label,
-                    'type': node.name,
-                    'object': object,
-                    'node': node
-                });
-            } else if (action === 'edit') {
-                target.closest('li').data('label', label).data('object', object);
-                target.closest('li').find('.itemLabel:first').text(label);
-                self.relocate();
+        this.$nestedModal.on('hide.bs.modal', function () { modalStateloading($(this)); });
+        this.$nestedModal.on('show.bs.modal', function () {
+            let $target =  $(this).data('target');
+            let action = $target.data('action'); //add or edit
+            let node = $target.data('node');
+            let nodeIcon = node.icon ? `<i class="${node.icon}"></i>` : '';
+            $(this).find('.modal-title').html(`${nodeIcon} <span>${action} ${node.label}</span>`);
+
+            let data = {};
+            if ('edit' === action) {
+                data = $target.closest('li').data('object');
+                data.label = $target.closest('li').data('label');
             }
 
-            nestedModal.modal('hide');
-        }
-
-        this.nestedModal = nestedModal;
-    }
+            self.ajaxNestedModal(node.url, JSON.stringify(data), 'application/json', function () {
+                modalStateActive(self.$nestedModal);
+            });
+        });
 
-    setNestedFormData(form, data)
-    {
-        function recursiveSetData(data, structure) {
-            for (let s in structure) {
-                if (typeof structure[s] === 'object' && data[s] !== null) {
-                    recursiveSetData(data[s], structure[s]);
-                } else {
-                    let element = form.find(`[name='${structure[s]}']`);
-                    let value = data && data.hasOwnProperty(s) ? data[s] : null;
+        this.$nestedModal.on('click', '.btn-json-menu-nested-save', function () {
+            let form = self.$nestedModal.find('form[name="json-menu-nested-form"]')
```

Comment:
IS it working with multiple json menu on the same page? It look like it applies on all intancies, But I may miss the bog picture here

Reply:
Fixed!